### PR TITLE
✨Source Twilio: New streams UserConversations with parent Users

### LINKED ISSUE #26938: Source Twilio : new stream user converstaions
### What area the feature impact?

Connectors

### Revelant Information

https://www.twilio.com/docs/conversations/api/user-conversation-resource#list-all-of-a-users-conversations

### PR DESCRIPTION
**Pull request by btkcodedev**
## What
Closes #26938

## Enhancement
New streams: 

- Users
- UserConversations

## Recommended reading order
1. `source_twilio/source.py`
2. `source_twilio/streams.py`
3.  `source_twilio/schemas/*`

## 🚨 User Impact 🚨
No breaking changes

## Pre-merge Checklist
<details open><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter [Yes]

- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [x] Issue acceptance criteria met
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [x] Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-twilio/source_twilio/streams.py [hunk 1]
```diff
@@ -611,3 +611,27 @@ def path(self, stream_slice: Mapping[str, Any], **kwargs):
...
+        return self.name.title()
+
+
+class UserConversations(TwilioNestedStream):
+    """https://www.twilio.com/docs/conversations/api/user-conversation-resource#list-all-of-a-users-conversations"""
+
+    parent_stream = Users
+    url_base = TWILIO_CONVERSATIONS_URL_BASE
+    uri_from_subresource = False
+    data_field = "conversations"
```
Comment (archangelic) [inline airbyte-integrations/connectors/source-twilio/source_twilio/streams.py:631]: It looks like this stream is currently inheriting a `primary_key` of `"sid"` but that is not a field in this stream. `primary_key` will need to be overwritten here.

**Review by archangelic — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-twilio/source_twilio/streams.py (hunk 1 above)
Comment (btkcodedev) [reply airbyte-integrations/connectors/source-twilio/source_twilio/streams.py:631]: @archangelic Primary key overwritten :)

**Comment by marcosmarxm** [comment]
@archangelic what is the status of this PR?

**Comment by archangelic** [comment]
I fixed some merge conflicts. Running tests now to see if we can get this merged asap. I will update soon.

**Review by archangelic — APPROVED** [review]

**Review by darynaishchenko — APPROVED** [review]

**Comment by darynaishchenko** [comment]
@btkcodedev The [acceptance tests](https://github.com/airbytehq/airbyte/pull/28073#issuecomment-1629092976) has started to fail, could you check it please.

**Pull request merged**